content/docs/azure/azureEndtoEnd.md

```diff
@@ -0,0 +1,171 @@
+
+![Client ID location](images/clientID2.PNG)
+
+Select “Certificates and Secrets” from the sidebar. Select “New client secret”. Give the client secret a description and select how long you would like it to remain active for. Once you click the “Add” button, make sure you take note of the client secret value and save it in a secure place. This is your service principal password.
+
+![Client secret location](images/password.PNG)
+
+### Add a Role Assignment
+Go to your resource group page on the Azure Portal. Select “Access control (IAM)” from the sidebar. Select “Add a role assignment”. Set the role to “Contributor” and search for the name you gave your app registration in the “Select” dropdown. Click “Save”. 
+![Role assignment](images/roleAssign.PNG)
+
+## Creating Containers from Docker Images
+### Install Docker
+You need to install Dokcer to be able to push and pull images to/from your Container registry.
+
+For Windows and WSL: [Guide](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly)
+
+For other OS: [Docker Desktop](https://hub.docker.com/?overlay=onboarding)
+
+### Build Images
+To deploy your code to Kubernetes, you must build your local project’s Docker images and push the containers to your Container Registry so that they are available in the cloud. 
+
+Set the path in Container Registry that you want to push the containers to:
+`export REGISTRY_PATH=<REGISTRY_NAME>.azurecr.io`
+
+Run the following command to authenticate your Container Registry: 
+`az acr login --name <REGISTRY_NAME>`
+
+Create a version, to be associated with your model each time it runs (change this accordingly):
+
+```export VERSION_TAG=1```
+
+Each docker image will be built and uploaded to the cloud using the Container Registry. 
+
+Note: If you would like to test a container locally, you can use the `docker run -it ${REGISTRY_PATH}<CONTAINER NAME>:$(VERSION_TAG}` before pushing to Container Registry. 
+	//Starting in the 'code' directory of azurepipeline
+
+	cd preprocess
+	docker build . -t ${REGISTRY_PATH}/preprocess:${VERSION_TAG}
+	docker push ${REGISTRY_PATH}/preprocess:${VERSION_TAG}
+
+	cd ../training
+	docker build . -t ${REGISTRY_PATH}/training:${VERSION_TAG}
+	docker push ${REGISTRY_PATH}/training:${VERSION_TAG}
+
+	cd ../register
+	docker build . -t ${REGISTRY_PATH}/register:${VERSION_TAG}
+	docker push ${REGISTRY_PATH}/register:${VERSION_TAG}
+
+	cd ../profile
+	docker build . -t ${REGISTRY_PATH}/profile:${VERSION_TAG}
+	docker push ${REGISTRY_PATH}/profile:${VERSION_TAG}
+
+	cd ../deploy
+	docker build . -t ${REGISTRY_PATH}/deploy:${VERSION_TAG}
+	docker push ${REGISTRY_PATH}/deploy:${VERSION_TAG}
+
+Once all of the images are pushed successfully, modify the pipeline.py file to use the appropriate image for each pipeline step.
```

Comment:
s/Once/When

("Once" is confusing to non-English speakers as it usually means less than twice.)